### PR TITLE
Fix overlapping MultiTextEdits causing conflicts

### PR DESCRIPTION
**Pull request by phase**
This takes all of the children in the `MutliTextEdit` we want to merge and makes
them children of the `MutliTextEdit` being merged into. Conflicts were occurring
when one MTE's edit range was overlapping with the others edit range.

I ran this over multiple codebases and this change didn't affect anything other
than the static imports that were failing.

### CODE REVIEW

**Comment by stephan-gh** [comment]
Can you add a test for a situation that was failing before and post how the "conflict" looks like (is it some exception)?

**Comment by phase** [comment]
Added a test case with a failing static import.

Here's the stacktrace:
```
Caused by: org.eclipse.text.edits.MalformedTreeException: Overlapping text edits
	at org.eclipse.text.edits.TextEdit$InsertionComparator.compare(TextEdit.java:126)
	at org.eclipse.text.edits.TextEdit$InsertionComparator.compare(TextEdit.java:1)
	at java.util.Collections.indexedBinarySearch(Collections.java:334)
	at java.util.Collections.binarySearch(Collections.java:322)
	at org.eclipse.text.edits.TextEdit.computeInsertionIndex(TextEdit.java:799)
	at org.eclipse.text.edits.TextEdit.internalAdd(TextEdit.java:784)
	at org.eclipse.text.edits.TextEdit.addChild(TextEdit.java:340)
	at org.cadixdev.mercury.RewriteContext.combineEdit(RewriteContext.java:105)
	at org.cadixdev.mercury.RewriteContext.rewrite(RewriteContext.java:88)
	at org.cadixdev.mercury.RewriteContext.process(RewriteContext.java:131)
	at org.cadixdev.mercury.Mercury.accept(Mercury.java:222)
```

File: src/test/resources/a/com/example/ImportTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package com.example;
+
+import com.example.other.AnotherClass;
+import static com.example.pkg.Constants.*;
+import com.example.other.OtherClass;
+import java.lang.String;
+import java.lang.Exception;
```
Comment (stephan-gh) [inline src/test/resources/a/com/example/ImportTest.java:17]: I think the problem is here: the static import is mixed with the other imports. Mercury handles normal imports through the `ImportRewrite`, but static imports through the normal remapping. That's why the `TextEdit`s overlap.

I suppose your workaround is fine for now if it avoids the problem, but in general we would also need to handle static imports through the `ImportRewrite`. Theoretically it is possible that you remap members in such a way that a remapped static import conflicts with some other remapped members in a class. Such conflicts are only handled within `ImportRewrite`.

Comment (jamierocks) [reply src/test/resources/a/com/example/ImportTest.java:17]: Do you think this is a reasonable "for-now" change to have merged?

Comment (stephan-gh) [reply src/test/resources/a/com/example/ImportTest.java:17]: It's probably fine, maybe with an extra comment that shows that this workaround should be removed again once someone manages to fix up all the import mess.

**Comment by jamierocks** [comment]
Merged with https://github.com/CadixDev/Mercury/commit/86723230ee16400ee7626e77c58a0bfdbeb02345.

**Pull request closed without merging**